packages/client/src/AdditionalApi.ts

```diff
@@ -123,104 +123,171 @@ export async function compressJSON(input: unknown): Promise<BodyInit> {
-        const container = builder.createString(ap.container)
-        const root = builder.createString(ap.root)
+    internLanguage(languageVersion: LionWebJsonUsedLanguage) : number {
+        if (!this.languagesIndexingMap.has(languageVersion)) {
+            const index = this.languages.length+1
+            this.languages.push({
+                key: this.internString(languageVersion.key),
+                version: this.internString(languageVersion.version)
+            } as PBLanguage)
+            return index
+        }
+        return this.languagesIndexingMap.get(languageVersion)
+    }
 
-        containerByAttached[ap.root] = ap.container
+    internMetaPointer(metaPointer: LionWebJsonMetaPointer) : number {
+        if (!this.metaPointersIndexingMap.has(metaPointer)) {
+            const index = this.metaPointers.length
+            this.metaPointers.push({
+                liLanguage: this.internLanguage({
+                    key: metaPointer.language,
+                    version: metaPointer.version
+                } as LionWebJsonMetaPointer),
+                siKey: this.internString(metaPointer.key)
+            } as PBMetaPointer)
+            return index
+        }
+        return this.metaPointersIndexingMap.get(metaPointer)
+    }
 
-        FBAttachPoint.startFBAttachPoint(builder)
-        FBAttachPoint.addContainer(builder, container)
-        FBAttachPoint.addContainment(builder, containment)
-        FBAttachPoint.addRoot(builder, root)
-        return FBAttachPoint.endFBAttachPoint(builder)
-    })
+    internString(string: string) : number {
+        if (string === undefined) {
+            return 0
+        }
+        if (!this.stringsIndexingMap.has(string)) {
+            const index = this.strings.length + 1
+            this.stringsIndexingMap.set(string, index)
+            this.strings.push(string)
+            return index
+        }
+        return this.stringsIndexingMap.get(string)
+    }
 
-    const attachPointsVector = FBBulkImport.createAttachPointsVector(builder, attachOffsets)
+}
 
-    const nodeOffsets = bulkImport.nodes.map(node => {
-        const propOffsets = node.properties.map(p => {
-            const mp = offsetForMetaPointer(builder, p.property)
-            const value = p.value ? builder.createString(p.value) : undefined
+/**
+ * Encodes a BulkImport object into a Protobuf byte array.
+ *
+ * The method processes each component of the BulkImport object, including attach points, nodes, properties,
+ * containments, references, and annotations, transforming them into their Protobuf representations while
+ * using the interning mechanism expected by protobuf.
+ *
+ * @param {BulkImport} bulkImport - The BulkImport object representing a collection of nodes and their structure to be serialized.
+ * @return {Uint8Array} A serialized Protobuf representation of the BulkImport object as a byte array.
+ */
+export function encodeBulkImportToProtobuf(bulkImport: BulkImport): Uint8Array {
```

Comment:
As this file is getting really long, I would prefer to put this function in a separate file.
Maybe together with the `InterningContext` class?